### PR TITLE
Add ipv6 matching support

### PR DESCRIPTION
**Pull request by Emantor**
I tried to keep the rule as simple as possible.

### CODE REVIEW

File: src/state.rs [hunk 1]
```diff
@@ -272,6 +273,19 @@ mod tests {
...
+  fn match_ipv6s() {
+    let lines = split("Lorem ipsum fe80::2:202:fe4 lorem\n Lorem 2001:67c:670:202:7ba8:5e41:1591:d723 lorem fe80::2:1 lorem ipsum fe80:22:312:fe::1%eth0");
+    let custom = [].to_vec();
+    let results = State::new(&lines, "abcd", &custom).matches(false, false);
+
+    assert_eq!(results.len(), 4);
+    assert_eq!(results.get(0).unwrap().text.clone(), "fe80::2:202:fe4");
+    assert_eq!(results.get(1).unwrap().text.clone(), "2001:67c:670:202:7ba8:5e41:1591:d723");
+    assert_eq!(results.get(2).unwrap().text.clone(), "fe80::2:1");
+    assert_eq!(results.get(3).unwrap().text.clone(), "fe80:22:312:fe::1%eth0");
```
Comment (fcsonline) [inline src/state.rs]: Oh!, I didn't know about this case. :+1:

**Comment by fcsonline** [comment]
Thanks for this PR! Great addition! 💯

**Comment by Emantor** [comment]
Getting through `rustfmt` turns out to be a bit churny :smile:

**Comment by Emantor** [comment]
Ping, is there anything left to do here for me?

**Comment by fcsonline** [comment]
Sorry! Merging!

**Pull request merged**